libs/clients/middlewares/README.md

```diff
@@ -0,0 +1,88 @@
+- Includes request timeout logic. By default, throws an error if there is no response in 10 seconds.
+- Throws an error for non-200 responses. The error object includes details from the response, including a problem property if the response implements the Problem Spec.
+- Logs circuit breaker events and information about failing requests.
+- Optionally opens the circuit for 400 responses.
+- Optionally parses and logs error response bodies.
+
+### Options
+
+- `name: string` - Name of fetch function. Used in logs and opossum stats.
+- `enableCircuitBreaker?: boolean` - Should use circuit breaker for requests. Defaults to `true`.
+- `timeout?: number | false` - Timeout for requests. Logged and thrown as errors. May cause circuit breaker to open. Defaults to `10000`ms. Can be disabled by passing false.
+- `treat400ResponsesAsErrors?: boolean` - If `true`, then too many 400 responses may cause the circuit to open. Either way these responses will be logged and thrown. Defaults to `false`.
+- `logErrorResponseBody?: boolean` - If `true`, then non-200 response bodies will be consumed and included in the error object and logged as `body`.
+- `opossum?: CircuitBreaker.Options` - Allows overriding Opossum options.
+
+The EnhancedFetch function works generally the same as standard fetch, except for non-200 responses it throws an error instead with the following properties.
+
+- `error.name: string` - "FetchError"
+- `error.response: Response` - The response object.
+- `error.url: string` - The requested url.
+- `error.status: number` - The response status code.
+- `error.statusText: string` - The response status text.
+- `error.headers: Headers` - The headers of the response.
+- `error.response: Response` - The response object. Body has not been consumed.
+- `error.problem?: object` - The parsed response body if the response has `content-type: application/problem+json`.
+- `error.problem?: object | string` - The response body if `logErrorResponseBody` was set to `true`. Parsed JSON or string depending on the response content type.
+
+### Examples
+
+```typescript
+import { createEnhancedFetch } from '@island.is/clients/middlewares'
+
+// Manual usage:
+const fetch = createEnhancedFetch({
+  name: 'my-fetch',
+  // Other options.
+})
+
+async function callApi() {
+  try {
+    const response = await fetch('/test')
+    const body = await response.json()
+  } catch (error) {
+    // For non-200 responses:
+    error.name // FetchError
+    error.response // the response object
+    error.problem // if server returned a standard problem response.
+    error.body // if configured with `logErrorResponseBody`.
+    error.url // the requested url
+    error.status // response status code
+    `error.statusText` // response status text
+    error.headers // response headers
```

Comment:
What's going on here?

Reply:
Ah. I wrote this to document how response errors look, but then wrote better docs further up. I'll remove it.